internal/services/cognitive/cognitive_deployment_resource.go

```diff
@@ -313,7 +312,20 @@ func (r CognitiveDeploymentResource) Delete() sdk.ResourceFunc {
 	return sdk.ResourceFunc{
 		Timeout: 30 * time.Minute,
 		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			var model cognitiveDeploymentModel
+			if err := metadata.Decode(&model); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
 			client := metadata.Client.Cognitive.DeploymentsClient
+			accountId, err := cognitiveservicesaccounts.ParseAccountID(model.CognitiveAccountId)
+
+			locks.ByID(accountId.ID())
+			defer locks.UnlockByID(accountId.ID())
+
+			if err != nil {
+				return err
+			}
```

Comment:
Unfortuntely, we can't rely on anything other than the ID being present here, however, we can build the account ID from the segments of the deployment ID:
```suggestion
			client := metadata.Client.Cognitive.DeploymentsClient

			accountId := cognitiveservicesaccounts.NewAccountID(id.SubscriptionId, id.ResourceGroupName, id.AccountName)
			locks.ByID(accountId.ID())
			defer locks.UnlockByID(accountId.ID())
```